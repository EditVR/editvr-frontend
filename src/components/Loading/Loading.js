/**
 * @file Loading.js
 * Exports a component that renders an indeterminate loading spinner.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { LinearProgress, Typography, withStyles } from '@material-ui/core';

import LoadingStyles from './Loading.style';

const Loading = ({ classes: { wrapper, spinner, title } }) =>
  <div className={wrapper}>
    <LinearProgress className={spinner} />
    <Typography className={title} type="display1">
      Loading...
    </Typography>
  </div>;

Loading.propTypes = {
  classes: PropTypes.shape({
    wrapper: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    spinner: PropTypes.string.isRequired
  }).isRequired
};

export default withStyles(LoadingStyles)(Loading);
