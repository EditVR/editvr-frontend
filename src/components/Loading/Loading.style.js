/**
 * @file Loading.style.js
 * Exports Loading component styles.
 */

export default theme => ({
  wrapper: {
    background: theme.palette.background.default,
    height: '100vh',
    maxHeight: '100%'
  },
  spinner: {
    flexGrow: 1
  },
  title: {
    paddingTop: 200,
    textAlign: 'center'
  }
});
