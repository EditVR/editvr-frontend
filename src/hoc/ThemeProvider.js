/**
 * @file ThemeProvider.js
 * Exports a HOC ThemeProvider component that attaches children their MUI theme.
 */

import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';

import muiTheme from '../styles/muiTheme';

const ThemeProvider = WrappedComponent => props => (
  <MuiThemeProvider theme={muiTheme}>
    <WrappedComponent {...props} />
  </MuiThemeProvider>
);

export default ThemeProvider;
