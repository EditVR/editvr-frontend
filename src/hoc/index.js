/**
 * @file index.js
 * Exports all hoc components.
 */

import ThemeProvider from './ThemeProvider/ThemeProvider';
import PrivateRoute from './PrivateRoute/PrivateRoute.container';
import PublicRoute from './PublicRoute/PublicRoute.container';

export { ThemeProvider, PrivateRoute, PublicRoute };
