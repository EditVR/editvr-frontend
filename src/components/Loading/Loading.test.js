/**
 * @file Loading.test.js
 * Contains tests for Loading.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from '../../hoc';
import Loading from './Loading';

const ThemedLoading = ThemeProvider(Loading);

describe('<Loading />', () => {
  it('Matches its snapshot', () => {
    expect(renderer.create(<ThemedLoading />).toJSON()).toMatchSnapshot();
  });
});
