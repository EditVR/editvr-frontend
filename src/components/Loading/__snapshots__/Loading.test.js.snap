// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Matches its snapshot 1`] = `
<div
  className="Loading-wrapper-1"
>
  <div
    className="MuiLinearProgress-root-4 MuiLinearProgress-colorPrimary-5 Loading-spinner-2"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar-12 MuiLinearProgress-barColorPrimary-13 MuiLinearProgress-bar1Indeterminate-15"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar-12 MuiLinearProgress-barColorPrimary-13 MuiLinearProgress-bar2Indeterminate-16"
      style={Object {}}
    />
  </div>
  <h1
    className="MuiTypography-root-20 MuiTypography-headline-25 Loading-title-3"
  >
    Loading...
  </h1>
</div>
`;
